pyaedt/edb_grpc/core/edb_data/padstacks_data.py

```diff
@@ -0,0 +1,1656 @@
+    edb_padstack :
+
+    ppadstack : str
+        Inherited AEDT object.
+
+    Examples
+    --------
+    >>> from pyaedt import Edb
+    >>> edb = Edb(myedb, edbversion="2021.2")
+    >>> edb_padstack = edb.core_padstack.definitions["MyPad"]
+    """
+
+    def __init__(self, edb_padstack, ppadstack):
+        self.edb_padstack = edb_padstack
+        self._ppadstack = ppadstack
+        self.pad_by_layer = {}
+        self.antipad_by_layer = {}
+        self.thermalpad_by_layer = {}
+        self._bounding_box = []
+        self._hole_params = None
+        for layer in self.via_layers:
+            self.pad_by_layer[layer] = EDBPadProperties(edb_padstack, layer, 0, self)
+            self.antipad_by_layer[layer] = EDBPadProperties(edb_padstack, layer, 1, self)
+            self.thermalpad_by_layer[layer] = EDBPadProperties(edb_padstack, layer, 2, self)
+        pass
+
+    @property
+    def name(self):
+        """Padstack Definition Name."""
+        return self.edb_padstack.GetName()
+
+    @property
+    def _padstack_methods(self):
+        return self._ppadstack._padstack_methods
+
+    @property
+    def _stackup_layers(self):
+        return self._ppadstack._stackup_layers
+
+    @property
+    def _builder(self):
+        return self._ppadstack._builder
+
+    @property
+    def _edb(self):
+        return self._ppadstack._edb
+
+    def _get_edb_value(self, value):
+        return self._ppadstack._get_edb_value(value)
+
+    @property
+    def via_layers(self):
+        """Layers.
+
+        Returns
+        -------
+        list
+            List of layers.
+        """
+        return self.edb_padstack.GetData().GetLayerNames()
+
+    @property
+    def via_start_layer(self):
+        """Starting layer.
+
+        Returns
+        -------
+        str
+            Name of the starting layer.
+        """
+        return list(self.via_layers)[0]
+
+    @property
+    def via_stop_layer(self):
+        """Stopping layer.
+
+        Returns
+        -------
+        str
+            Name of the stopping layer.
+        """
+        return list(self.via_layers)[-1]
+
+    @property
+    def hole_params(self):
+        """Via Hole parameters values."""
+
+        viaData = self.edb_padstack.GetData()
+        self._hole_params = viaData.GetHoleParametersValue()
+        return self._hole_params
+
+    @property
+    def hole_parameters(self):
+        """Hole parameters.
+
+        Returns
+        -------
+        list
+            List of the hole parameters.
+        """
+        self._hole_parameters = self.hole_params[2]
+        return self._hole_parameters
+
+    @pyaedt_function_handler()
+    def _update_hole_parameters(self, hole_type=None, params=None, offsetx=None, offsety=None, rotation=None):
+        """Update hole parameters.
+
+        Parameters
+        ----------
+        hole_type : optional
+            Type of the hole. The default is ``None``.
+        params : optional
+            The default is ``None``.
```

Comment:
```suggestion
        hole_type : optional
            Type of the hole. The default is ``None``.
        params : optional
            The default is ``None``.
```
```suggestion
        hole_type : optional
            Type of the hole. The default is ``None``.
        params : list, optional
            List of hole parameters to update. The default is ``None``.
```
Datatype needed for hole_type.